mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java

```diff
@@ -74,14 +104,22 @@ public DeleteControl deleteResource(ServiceDomain sd, Context<ServiceDomain> con
             createOrUpdateDeployment(sd);
             createOrUpdateService(sd);
             if(ServiceDomainSpec.ExposeType.Http == sd.getSpec().getExpose()) {
-                createOrUpdateCamelKIntegration(sd);
+                final String sdConfigMapName = "integration-" + sd.getSpec().getType().getTypeAsString() + "-http";
+                ConfigMap sdConfigMap = client.configMaps().inNamespace(sd.getMetadata().getNamespace()).withName(sdConfigMapName).get();
+
+                final boolean validateSdConfigMap = validateSdConfigMap(sd, sdConfigMapName, sdConfigMap);
+                if(!validateSdConfigMap){
+                    return UpdateControl.noUpdate();
+                }
+
+                createOrUpdateCamelKIntegration(sd, sdConfigMap);
             } else {
                 deleteCamelIntegration(sd);
             }
             String kafkaTopic = createKafkaTopic(sd);
-            String kafkaUser = createKafkaUser(sd, kafkaTopic);
+//            String kafkaUser = createKafkaUser(sd, kafkaTopic);
             status.setKafkaTopic(kafkaTopic);
-            status.setKafkaUser(kafkaUser);
+//            status.setKafkaUser(kafkaUser);
         } catch (Exception e) {
             LOGGER.error("{} service domain failed to be created/updated", sdName, e);
```

Comment:
If this fails I'd expect the status to be updated with some message

Reply:
Done